packages/components/pie-modal/src/index.ts

```diff
@@ -154,33 +154,60 @@ export class PieModal extends RtlMixin(LitElement) implements ModalProps {
         }
     }
 
-    render () {
-        const {
-            heading,
-            headingLevel = 'h2',
-            size,
-            isFullWidthBelowMid,
-        } = this;
+    public render = () => html`
+        <dialog
+            id="dialog"
+            class="c-modal"
+            size="${this.size}"
+            ?isFullWidthBelowMid=${this.isFullWidthBelowMid}
+            data-test-id="pie-modal">
+            <header>
+                ${this._renderModalHeader()}
+            </header>
+            <article class="c-modal-content c-modal-content--scrollable">
+                <slot></slot>
+            </article>
+            <footer class="c-modal-footer">
+                ${this._renderModalFooter()}
+            </footer>
+        </dialog>`;
 
-        const headingTag = unsafeStatic(headingLevel);
+    /**
+     * Returns the HTML content for the header.
+     * Extracted to keep the render method succinct.
+     * @returns {TemplateResult} HTML template for the modal's header
+     */
+    private _renderModalHeader () : TemplateResult {
+        const headingTag = unsafeStatic(this.headingLevel);
 
         return html`
-            <dialog
-                id="dialog"
-                class="c-modal"
-                size="${size}"
-                ?isFullWidthBelowMid=${isFullWidthBelowMid}>
-                <header>
-                    <${headingTag} class="c-modal-heading">${heading}</${headingTag}>
-                     ${this.isDismissible ? this.renderCloseButton() : nothing}
-                </header>
-                <article class="c-modal-content">
-                    <slot></slot>
-                </article>
-            </dialog>
-        `;
+            <${headingTag} class="c-modal-heading">
+                ${this.heading}
+            </${headingTag}>
+            ${this.isDismissible ? this.renderCloseButton() : nothing}`;
     }
 
+    /**
+     * Returns the HTML content for the footer.
+     * Extracted to keep the render method succinct.
+     * @returns {TemplateResult} HTML template for the modal's footer
+    */
+    private _renderModalFooter = () : TemplateResult => html`
+        <pie-button
+            variant="primary"
+            type="submit"
+            @click="${() => this._dialog?.close('leading')}"
+            data-test-id="modal-leading-action">
+            Confirm
```

Comment:
should these be user provided?

Reply:
Yes, I've now made a ticket to cover copy and i18n in general